crates/kv-store/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+//! 3. Write the number of key-value pairs.
+//! 4. **Compress** the entire block using LZ4.
+//! 5. Calculate and append xxhash_32 checksum.
+//!
+//! Decoding:
+//! 1. Verify the xxhash_32 checksum.
+//! 2. **Decompress** the block using LZ4.
+//! 3. Read the number of key-value pairs.
+//! 4. Read offsets for each key-value pair.
+//! 5. Parse individual key-value chunks.
+//!
+//! ### Large Value Block
+//!
+//! Large Value Blocks store a single key-value pair with a large value.
+//!
+//! ┌──────────────────────────┐
+//! │Large Block               │
+//! │┌ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ─ ─ ─ │
+//! │  value   Block Checksum ││
+//! ││ bytes │      u32        │
+//! │ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘│
+//! └──────────────────────────┘
+//!
+//! Encoding:
+//! 1. Write the value bytes.
+//! 2. Calculate and append xxhash_32 checksum.
+//!
+//! Decoding:
+//! 1. Verify the xxhash_32 checksum.
+//! 2. Read the value bytes.
+//!
+//! We need not encode the length of value, because we can get the whole Block by offset in meta.
+//!
+//! ## Block Meta
+//!
+//! The Block Meta section contains metadata for all blocks in the SSTable.
+//!
+//! ┌────────────────────────────────────────────────────────────┐
+//! │ All Block Meta                                             │
+//! │┌ ─ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─┌ ─ ─ ─┌ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ │
+//! │  block length │ Block Meta │ ...  │ Block Meta │ checksum ││
+//! ││     u32      │   bytes    │      │   bytes    │   u32     │
+//! │ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ┘─ ─ ─ ┘─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ┘│
+//! └────────────────────────────────────────────────────────────┘
+//!
+//! Each Block Meta entry is encoded as follows:
+//!
+//! ┌──────────────────────────────────────────────────────────────────────────────────────────────┐
+//! │ Block Meta                                                                                   │
+//! │┌ ─ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ─ ┐ │
+//! │  block offset │ first key len   first key   large & compress │ last key len     last key     │
+//! ││     u32      │      u16      │   bytes   │        u8        │  u16(option)  │bytes(option)│ │
+//! │ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─  │
+//! └──────────────────────────────────────────────────────────────────────────────────────────────┘
+//!
+//! Encoding:
+//! 1. Write the number of blocks.
+//! 2. For each block, write its metadata (offset, first key, is_large and compression_type flag, and last key if not large).
+//!     - is_large and compression_type flag: the first bit for is_large, the next 7 bits for compression_type.
+//! 3. Calculate and append xxhash_32 checksum.
+//!
+//! Decoding:
+//! 1. Read the number of blocks.
+//! 2. For each block, read its metadata.
+//! 3. Verify the xxhash_32 checksum.
+//!
+//!
+//! Note: In this crate, **only** [MemStoreIterator] will filter empty value,
```

Comment:
What's the difference between filter empty value and use empty value as deletetion?

Reply:
Changed to 
```
//! Note: In this crate, the empty value is regarded as deleted. **only** [MemStoreIterator] will filter empty value.
//! Other iterators will still return empty value.
```